hyp3_sdk/jobs.py

```diff
@@ -1,96 +1,221 @@
 
+    def complete(self) -> bool:
+        return self.status_code in ('FAILED', 'SUCCEEDED')
 
-def make_autorift_job(job_name: str, granule1: str, granule2: str) -> Job:
-    """Make an autoRIFT Job object
+    def succeeded(self) -> bool:
+        return self.status_code == 'SUCCEEDED'
 
-    Args:
-        job_name: A name for the job (must be <= 20 characters)
-        granule1: The first granule (scene) to use
-        granule2: The second granule (scene) to use
+    def failed(self) -> bool:
+        return self.status_code == 'FAILED'
 
-    Returns:
-        A Job object
-    """
-    return Job('AUTORIFT', job_name, {'granules': [granule1, granule2]})
+    def running(self) -> bool:
+        return self.status_code in ('PENDING', 'RUNNING')
 
+    def expired(self) -> bool:
+        try:
+            return datetime.now(tz.UTC) >= self.expiration_time
+        except TypeError:
+            raise HyP3Error('Only SUCCEEDED jobs have an expiration time')
 
-def make_rtc_gamma_job(job_name: str, granule: str, extra_parameters: dict = {}) -> Job:
-    """Make an RTC Job object
+    # TODO: handle expired products
+    def download_files(self, location: Path) -> List[Path]:
+        """
+        Args:
+            location: Directory location to put files into
 
-    Args:
-        job_name: A name for the job (must be <= 20 characters)
-        granule: The granule (scene) to process
-        extra_parameters: Extra job parameters specifying custom processing options
+        Returns: list of Path objects to downloaded files
+        """
+        if not self.complete():
+            raise HyP3Error('Incomplete jobs cannot be downloaded')
+        downloaded_files = []
+        for file in self.files:
+            download_url = file['url']
+            filename = location / file['filename']
+            try:
+                downloaded_files.append(download_file(download_url, filename))
+            except Exception:
+                raise HyP3Error('unable to download file')
+        return downloaded_files
+
+
+class Batch:
+    def __init__(self, jobs: List[Job]):
+        if len(jobs) == 0:
+            warnings.warn('Jobs list is empty; creating an empty Batch', UserWarning)
+
+        self.jobs = jobs
+
+    def __len__(self):
+        return len(self.jobs)
+
+    def __add__(self, other: Union[Job, 'Batch']):
+        if isinstance(other, Batch):
+            return Batch(self.jobs + other.jobs)
+        elif isinstance(other, Job):
+            return Batch(self.jobs + [other])
+        else:
+            raise TypeError(f"unsupported operand type(s) for +: '{type(self)}' and '{type(other)}'")
+
+    def __repr__(self):
+        return str([job.to_dict() for job in self.jobs])
+
+    def complete(self) -> bool:
+        """
+        Returns: True if all jobs are complete, otherwise returns False
+        """
+        for job in self.jobs:
+            if not job.complete():
+                return False
+        return True
```

Comment:
tempted to do `return all(job.complete() for job in jobs)`, I think it's still readable, but I might be too excited for a chance to use python functions I don't usually